src/Components/Search/search.tsx

```diff
@@ -1,52 +1,56 @@
-import React, { ChangeEvent, FC, ReactElement, useState } from 'react';
+import React, { FC, ReactElement, useCallback, useState } from 'react';
 import { Button } from '../button/button';
 import { SearchBy } from '../app/app.types';
 import { SearchProps } from './search.types';
 import './search.style.scss';
+import { useDispatch, useSelector } from 'react-redux';
+import { clickOnSearch, toggleSearchOption } from '../../store/movies/movies.actions';
 
-export const Search: FC<SearchProps> = ({
-    title,
-    onSearch,
-    searchBy,
-    onSearchByChange,
-}): ReactElement => {
-    const [value, setValue] = useState('');
+export const Search: FC<SearchProps> = ({ title }): ReactElement => {
+    const [inputSearch, setInputSearch] = useState('');
+    const searchBy = useSelector((store) => store.searchBy);
+    const dispatch = useDispatch();
 
-    const handleSearchByButtonClick = (searchByValue: SearchBy) => (): void => {
-        onSearchByChange(searchByValue);
-    };
+    console.log(inputSearch);
+
+    const handleInputChange = (event) => setInputSearch(event.target.value);
 
-    const handleInputChange = (event: ChangeEvent<HTMLInputElement>) =>
-        setValue(event.target.value);
+    const toggleSearchBy = (searchBy: SearchBy) =>
+        useCallback(() => {
+            dispatch(toggleSearchOption(searchBy));
+        }, [searchBy]);
```

Comment:
It does not work. React hooks should locate in a render function.

Memoization hooks for carrying are unusable.